File: prog sys/tp2/Producteur.java
import java.util.Random;

class Producteur extends Thread {

    private Donnee d;

    public Producteur(Donnee d){
        this.d=d;
    }

    @Override
    public void run() {
        Random rnd = new Random();
        for (int i = 1 ; i<1000 ; i++){
            try {
                String c =String.valueOf((char)('a' + rnd.nextInt(26)));
                d.ajouter(c);
                Thread.sleep(1000);
            } catch (InterruptedException e){
                System.err.println(e.getMessage());
            }
            

        }
        
    }
} 

File: prog sys/td5/cours-python/2_ProdCons/0_VersionObjet/Producteur.py
import threading
import time
import random

class Producteur(threading.Thread):
    def __init__(self, donnee, nom):
        super().__init__()
        self.donnee = donnee
        self.nom = nom

    def run(self):
        while True:
            item = random.randint(1, 100)  # Générer un élément aléatoire
            print(f"[{self.nom}] Produit {item}")
            self.donnee.produire(self.nom, item)
            time.sleep(random.uniform(0.5, 2))  # Simuler un délai